src/include/catalog/postgres/pg_statistic.h

```diff
@@ -30,13 +36,42 @@ class PgStatistic {
    */
   static constexpr CatalogColumnDef<table_oid_t, uint32_t> STARELID{col_oid_t{1}};
   static constexpr CatalogColumnDef<col_oid_t, uint32_t> STAATTNUM{col_oid_t{2}};
-  static constexpr CatalogColumnDef<uint32_t, uint32_t> STA_NULLROWS{col_oid_t{3}};
-  static constexpr CatalogColumnDef<uint32_t, uint32_t> STA_NUMROWS{col_oid_t{4}};
+  static constexpr CatalogColumnDef<uint32_t, uint32_t> STA_NUMROWS{col_oid_t{3}};
+  static constexpr CatalogColumnDef<uint32_t, uint32_t> STA_NONNULLROWS{col_oid_t{4}};
+  static constexpr CatalogColumnDef<uint32_t, uint32_t> STA_DISTINCTROWS{col_oid_t{5}};
+  static constexpr CatalogColumnDef<storage::VarlenEntry> STA_TOPK{col_oid_t{6}};
+  static constexpr CatalogColumnDef<storage::VarlenEntry> STA_HISTOGRAM{col_oid_t{7}};
 
-  static constexpr uint8_t NUM_PG_STATISTIC_COLS = 4;
+  static constexpr uint8_t NUM_PG_STATISTIC_COLS = 7;
 
   static constexpr std::array<col_oid_t, NUM_PG_STATISTIC_COLS> PG_STATISTIC_ALL_COL_OIDS = {
-      STARELID.oid_, STAATTNUM.oid_, STA_NULLROWS.oid_, STA_NUMROWS.oid_};
+      STARELID.oid_,         STAATTNUM.oid_, STA_NUMROWS.oid_,  STA_NONNULLROWS.oid_,
+      STA_DISTINCTROWS.oid_, STA_TOPK.oid_,  STA_HISTOGRAM.oid_};
+
+ public:
+  /**
+   * Contains information on how to derive values for the columns within pg_statistic
+   */
+  struct PgStatisticColInfo {
+    /** The type of aggregate to use */
+    parser::ExpressionType aggregate_type_;
+    /** Whether the aggregate is distinct */
+    bool distinct_;
+    /** Oid of the column */
+    catalog::col_oid_t column_oid_;
+    /** Human readable name of the column */
+    std::string_view name_;
+  };
+  static constexpr uint8_t NUM_ANALYZE_AGGREGATES = 4;
+  static constexpr std::array<PgStatisticColInfo, NUM_ANALYZE_AGGREGATES> ANALYZE_AGGREGATES = {
```

Comment:
I'm not sure if this is the best place for this. It's used by `AnalyzeTranslator`, and `InputColumnDeriver`. If you have a better suggestion please let me know.

Reply:
Maybe if I give some background on to how this is used and why I put it here we can find a better place. The reason I created this object was to try and make it as easy as possible to add columns to `pg_statistic` and try and reduce to number of files that need to be changed. Right now the process looks something like this
1. Add column details to this class
2. Add column details to pg_statistic_impl
3. Add a new entry to this array

The reason I finally settled on this class is because this array is basically describing how to populate the pg_statistic table.

Maybe pg_statistic_impl.h is better since this is more of an implementation detail?